3-networks/modules/standard_shared_vpc/main.tf

```diff
@@ -111,3 +154,67 @@ resource "google_compute_router_nat" "default_nat" {
     enable = true
   }
 }
+
+module "region1_router1" {
+  source  = "terraform-google-modules/cloud-router/google"
+  name    = "cr-${local.vpc_name}-${local.subnets[0].subnet_region}-cr1"
+  project = var.project_id
+  network = module.main.network_name
+  region  = local.subnets[0].subnet_region
```

Comment:
I think the regions should be provided explicitly as locals/variables, rather than being retrieved from subnets. (you might have multiple subnets in one region)